packages/camera/camera_android_camerax/test/live_data_test.dart

```diff
@@ -0,0 +1,164 @@
+
+import 'live_data_test.mocks.dart';
+import 'test_camerax_library.g.dart';
+
+@GenerateMocks(<Type>[TestLiveDataHostApi, TestInstanceManagerHostApi])
+void main() {
+  TestWidgetsFlutterBinding.ensureInitialized();
+
+  // Mocks the call to clear the native InstanceManager.
+  TestInstanceManagerHostApi.setup(MockTestInstanceManagerHostApi());
+
+  group('LiveData', () {
+    tearDown(() {
+      TestLiveDataHostApi.setup(null);
+      TestInstanceManagerHostApi.setup(null);
+    });
+
+    test('observe makes call to add observer to LiveData instance', () async {
+      final MockTestLiveDataHostApi mockApi = MockTestLiveDataHostApi();
+      TestLiveDataHostApi.setup(mockApi);
+
+      final InstanceManager instanceManager = InstanceManager(
+        onWeakReferenceRemoved: (_) {},
+      );
+
+      final LiveData<Object> instance = LiveData<Object>.detached(
+        instanceManager: instanceManager,
+      );
+      const int instanceIdentifier = 0;
+      instanceManager.addHostCreatedInstance(
+        instance,
+        instanceIdentifier,
+        onCopy: (LiveData<Object> original) => LiveData<Object>.detached(
+          instanceManager: instanceManager,
+        ),
+      );
+
+      final Observer<Object> observer = Observer<Object>.detached(
+        instanceManager: instanceManager,
+        onChanged: (Object value) {},
+      );
+      const int observerIdentifier = 20;
+      instanceManager.addHostCreatedInstance(
+        observer,
+        observerIdentifier,
+        onCopy: (_) => Observer<Object>.detached(
+          instanceManager: instanceManager,
+          onChanged: (Object value) {},
+        ),
+      );
+
+      await instance.observe(
+        observer,
+      );
+
+      verify(mockApi.observe(
+        instanceIdentifier,
+        observerIdentifier,
+      ));
+    });
+
+    test(
+        'removeObservers makes call to remove observers from LiveData instance',
+        () async {
+      final MockTestLiveDataHostApi mockApi = MockTestLiveDataHostApi();
+      TestLiveDataHostApi.setup(mockApi);
+
+      final InstanceManager instanceManager = InstanceManager(
+        onWeakReferenceRemoved: (_) {},
+      );
+
+      final LiveData<Object> instance = LiveData<Object>.detached(
+        instanceManager: instanceManager,
+      );
+      const int instanceIdentifier = 0;
+      instanceManager.addHostCreatedInstance(
+        instance,
+        instanceIdentifier,
+        onCopy: (LiveData<Object> original) => LiveData<Object>.detached(
+          instanceManager: instanceManager,
+        ),
+      );
+
+      await instance.removeObservers();
+
+      verify(mockApi.removeObservers(
+        instanceIdentifier,
+      ));
+    });
+
+    test('getValue returns expected value', () async {
+      final MockTestLiveDataHostApi mockApi = MockTestLiveDataHostApi();
+      TestLiveDataHostApi.setup(mockApi);
+      TestInstanceManagerHostApi.setup(MockTestInstanceManagerHostApi());
```

Comment:
@bparrishMines I ended up needing to set this up here because setting it at the top of the main method didn't work, but it also didn't work without it at the top of the file. Do you know why or is there some technical details I'm missing?